### ISSUE TITLE
[3D Model] 3D model described as an image

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#868]
## Issue summary

When a 3D model is discovered by a screen reader user, the model is described as an "image" or "graphic" via the current `role` attribute value. This value provides incorrect semantic meaning and context which could be confusing for screen reader users.

In addition, Windows based screen reader users are unable to interact with a 3D model. This is due to screen readers such as JAWS, NVDA, or Windows Narrator using single arrow keypress events to navigate page content.

<details>
  <summary>Screenshots</summary>

  <img width="1189" alt="Dawn product page. A 3D model of an astronaut is highlighted. VoiceOver output window shows the text, Test astronaut model, image." src="https://user-images.githubusercontent.com/1392632/141175783-7ce61493-7258-46f1-82d4-ecafb3b7d83e.png">

</details>

#### HTML

```html
<div
  class="userInput"
  tabindex="0"
  role="img"
  aria-label="…"
  aria-live="polite"
  style="cursor: grab"
>
  <!--- Canvas -->
</div>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. Navigate to a product with one or more 3D models available
3. Activate the 3D model activtor control
4. With an active screen reader, use the virtual cursor to navigate to the 3D model

### Expected

- 3D model element to be conveyed as a "3D model"

### Actual

- 3D model is conveyed as an image element

## Recommendation

1. Set the `role` attribute to `application`. This will allow the 3D model to override assistive technology defaults to accept arrow key events (by default, Windows based screen readers use arrow keys to move the virtual cursor.)
2. To set a more canonical role description, add the `aria-roledescription` attribute, setting its value to `3D Model`.

With this, the screen reader would announce,
> _"{alt}, 3D Model"_

Review the post "[3D Model Accessibility](https://scottvinkle.me/blogs/work/3d-model-accessibility)" for more details on why a 3D model is _not_ an image (or video content.)

#### HTML

```html
<div
  class="userInput"
  tabindex="0"
  role="application"
  aria-roledescription="3D Model"
  aria-label="[{string} description of 3D model]"
  aria-describedby="[{IDREF} containing information on how to interact with the 3D model]"
  style="cursor: grab"
>
  <!--- Canvas -->
</div>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.2.0
- **Component:** 3D Model
- **WCAG Principal:** Perceivable, Robust
- **WCAG SC:** [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html), [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
- **Severity:** High
- **Effort:** Low
- **Related:** https://github.com/google/model-viewer/issues/443

**Comment by svinkle** [comment]
@ludoboludo Thanks. Some of the 3d model tickets will be out of scope. I wanted to have a central location for communication between teams. Could we ping some folks here to get more 👀?

**Comment by JonathanWade** [comment]
@elalish Are these recommendations something that should land at the <model-viewer> level instead of attributes Shopify tacks on? Feels like users outside our ecosystem would benefit as well.

**Comment by elalish** [comment]
Agreed that this sounds like a MV-level issue. We're starting  to get some internal customers now too and with that more a11y scrutiny. Might be good to get together and see if we can make more of an a11y plan. 

One concern that  was brought up internally was that the progress/loading screen reader stuff gets distracting. Curious if anyone on your  side has thoughts  (might be a bigger deal with several models on the page).

**Comment by svinkle** [comment]
@elalish One could make the argument the initial loading sequence of messages shared to the user are a little verbose. In my testing of multiple models on a single page, this has not affected the user experience as only one model loads at a time. However, when the user becomes familiar with how `model-viewer` works and functions, these could be considered a nuisance.

I've made recommendations in the past on tidying up the DOM and organizing live region announcements in https://github.com/google/model-viewer/issues/462, https://github.com/google/model-viewer/issues/461, and https://github.com/google/model-viewer/issues/445. However, I don't believe this to be as "mission critical" as the usability barriers found in https://github.com/Shopify/dawn/issues/869 or this issue reported here.

To @JonathanWade's point, this would be a `model-viewer` level change which would positively affect other instances of the component.

**Comment by elalish** [comment]
Sorry, I would have dealt with this earlier if I'd remembered. Is there a MV issue for this anywhere? Unfortunately I'm in the process of cutting a release right now, so this will have to wait. But I'd be happy to assist.

**Comment by svinkle** [comment]
@elalish No worries. Let's use the time during the next release to address and thoroughly test this particular issue.

There's an MV issue in https://github.com/google/model-viewer/issues/443 but this ticket has more details.
